File: frontend/src/types/job.ts
export interface Job {
  id: number;
  title: string;
  company: string;
  description: string;
  location: string | null;
  status: string;
  created_at: string;
}

export interface JobCreate {
  title: string;
  company: string;
  description: string;
  location?: string;
  status?: string;
}

export interface JobUpdate {
  title?: string;
  company?: string;
  description?: string;
  location?: string;
  status?: string;
}

File: backend/app/schemas/job.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class JobCreate(BaseModel):
    title: str = Field(min_length=1, max_length=255)
    company: str = Field(min_length=1, max_length=255)
    description: str = Field(min_length=1)
    location: str | None = Field(default=None, max_length=255)
    status: str = Field(default="applied", max_length=50)


class JobUpdate(BaseModel):
    title: str | None = Field(default=None, min_length=1, max_length=255)
    company: str | None = Field(default=None, min_length=1, max_length=255)
    description: str | None = Field(default=None, min_length=1)
    location: str | None = Field(default=None, max_length=255)
    status: str | None = Field(default=None, max_length=50)


class JobResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    title: str
    company: str
    description: str
    location: str | None
    status: str
    created_at: datetime
